glyphs-reader/src/propagate_anchors.rs

```diff
@@ -0,0 +1,883 @@
+    }
+
+    #[test]
+    fn origin_anchor() {
+        // derived from the observed behaviour of glyphs 3.2.2 (3259)
+        let mut glyphs = GlyphSetBuilder::new()
+            .add_glyph("a", |glyph| {
+                glyph
+                    .add_anchor("*origin", (-20, 0))
+                    .add_anchor("bottom", (242, 7))
+                    .add_anchor("ogonek", (402, 9))
+                    .add_anchor("top", (246, 548));
+            })
+            .add_glyph("acutecomb", |glyph| {
+                glyph
+                    .add_anchor("_top", (150, 580))
+                    .add_anchor("top", (170, 792));
+            })
+            .add_glyph("aa", |glyph| {
+                glyph
+                    .add_anchor("bottom_1", (218, 8))
+                    .add_anchor("bottom_2", (742, 7))
+                    .add_anchor("ogonek_1", (398, 9))
+                    .add_anchor("ogonek_2", (902, 9))
+                    .add_anchor("top_1", (227, 548))
+                    .add_anchor("top_2", (746, 548));
+            })
+            .add_glyph("a_a", |glyph| {
+                glyph.add_component("aa", (0, 0));
+            })
+            .add_glyph("a_aacute", |glyph| {
+                glyph
+                    .add_component("a_a", (0, 0))
+                    .add_component("acutecomb", (596, -32))
+                    .add_component_anchor("top_2");
+            })
+            .build();
+        propagate_all_anchors_impl(&mut glyphs);
+
+        let new_glyph = glyphs.get("a_aacute").unwrap();
+        assert_eq!(
+            new_glyph.layers[0].anchors,
+            [
+                ("bottom_1", (218., 8.)),
+                ("bottom_2", (742., 7.)),
+                ("ogonek_1", (398., 9.)),
+                ("ogonek_2", (902., 9.)),
+                ("top_1", (227., 548.)),
+                ("top_2", (766., 760.)),
+            ]
+        );
+    }
+
+    #[test]
+    fn invert_names_on_rotation() {
+        // derived from the observed behaviour of glyphs 3.2.2 (3259)
+        let mut glyphs = GlyphSetBuilder::new()
+            .add_glyph("comma", |_| {})
+            .add_glyph("commaaccentcomb", |glyph| {
+                glyph
+                    .add_anchor("_bottom", (289, 0))
+                    .add_anchor("mybottom", (277, -308))
+                    .add_component("comma", (9, -164));
+            })
+            .add_glyph("commaturnedabovecomb", |glyph| {
+                glyph
+                    .add_component("commaaccentcomb", (589, 502))
+                    .rotate_component(180.);
+            })
+            .build();
+        propagate_all_anchors_impl(&mut glyphs);
+
+        let new_glyph = glyphs.get("commaturnedabovecomb").unwrap();
+        assert_eq!(
+            new_glyph.layers[0].anchors,
+            [("_top", (300., 502.)), ("mytop", (312., 810.)),]
+        );
+    }
+
+    #[test]
+    fn affine_scale() {
```

Comment:
This looks like a bunch of independent tests that don't need to be one big test?

Reply:
just me sanity checking the behaviour of this fn, doesn't feel like much of an improvement to break this into five two line tests..